File: internal/content/testdata/bench/go/elements-appearing-more-than-n-over-3.go
import "sort"

func majority_elements_n3(nums []int) []int {
	cnt1, cnt2 := 0, 0
	var num1, num2 *int
	for _, n := range nums {
		nCopy := n
		if num1 != nil && n == *num1 {
			cnt1++
		} else if num2 != nil && n == *num2 {
			cnt2++
		} else if cnt1 == 0 {
			num1 = &nCopy
			cnt1 = 1
		} else if cnt2 == 0 {
			num2 = &nCopy
			cnt2 = 1
		} else {
			cnt1--
			cnt2--
		}
	}
	threshold := len(nums) / 3
	result := []int{}
	if num1 != nil {
		count := 0
		for _, x := range nums {
			if x == *num1 {
				count++
			}
		}
		if count > threshold {
			result = append(result, *num1)
		}
	}
	if num2 != nil && (num1 == nil || *num2 != *num1) {
		count := 0
		for _, x := range nums {
			if x == *num2 {
				count++
			}
		}
		if count > threshold {
			result = append(result, *num2)
		}
	}
	sort.Ints(result)
	return result
}

File: internal/content/testdata/bench/java/elements-appearing-more-than-n-over-3.java
import java.util.*;

class Solution {
    public long[] majority_elements_n3(long[] nums) {
        long cnt1 = 0, cnt2 = 0;
        Long num1 = null, num2 = null;
        for (long n : nums) {
            if (num1 != null && n == num1) {
                cnt1++;
            } else if (num2 != null && n == num2) {
                cnt2++;
            } else if (cnt1 == 0) {
                num1 = n;
                cnt1 = 1;
            } else if (cnt2 == 0) {
                num2 = n;
                cnt2 = 1;
            } else {
                cnt1--;
                cnt2--;
            }
        }
        long threshold = nums.length / 3;
        List<Long> result = new ArrayList<>();
        if (num1 != null) {
            long count = 0;
            for (long x : nums) if (x == num1) count++;
            if (count > threshold) result.add(num1);
        }
        if (num2 != null && !num2.equals(num1)) {
            long count = 0;
            for (long x : nums) if (x == num2) count++;
            if (count > threshold) result.add(num2);
        }
        Collections.sort(result);
        long[] arr = new long[result.size()];
        for (int i = 0; i < result.size(); i++) arr[i] = result.get(i);
        return arr;
    }
}
